packages/matchbox/src/components/Modal/Modal.js

```diff
@@ -1,88 +1,144 @@
+      onKey('escape', onClose)(e);
+    }
   };
 
-  handleOutsideClick = (e) => {
-    const { open, onClose } = this.props;
-    const isOutside = this.content && !this.content.contains(e.target) && this.container && this.container.contains(e.target);
+  const handleOutsideClick = e => {
+    const { open } = props;
+    const isOutside =
+      content && !content.contains(e.target) && container && container.contains(e.target);
 
     if (open && isOutside && onClose) {
       onClose(e);
     }
-  }
+  };
 
-  handleKeyDown = (e) => {
-    const { onClose, open } = this.props;
+  return (
+    <Portal containerId={containerId}>
+      <StyledBase
+        {...rest}
+        className={className}
+        onClose={onClose}
+        ref={el => (container = el)}
+        role="dialog"
+        aria-modal="true"
+      >
+        <StyledWrapper>
+          <ModalContent open={props.open} maxWidth={props.maxWidth}>
+            <div ref={el => (content = el)}>
+              <WindowEvent event="keydown" handler={handleKeydown} />
+
+              <WindowEvent event="click" handler={handleOutsideClick} />
 
-    if (open && onClose) {
-      onKey('escape', onClose)(e);
-    }
-  }
-
-  render() {
-    const {
-      onClose,
-      open,
-      children,
-      className,
-      showCloseButton,
-      ...rest
-    } = this.props;
-
-    const modalClasses = classnames(
-      styles.Modal,
-      open && styles.open,
-      className
-    );
-
-    return (
-      <div className={modalClasses} onClose={onClose} {...rest} ref={(node) => this.container = node} role="dialog" aria-modal="true">
-        <Grid center='xs' middle='xs' className={styles.Grid}>
-          <Grid.Column xs={11} md={9} xl={7}>
-            <Content contentRef={(node) => this.content = node} open={open}>
-              <WindowEvent event='keydown' handler={this.handleKeyDown} />
-              <WindowEvent event='click' handler={this.handleOutsideClick} />
               {showCloseButton && (
-                <Button className={styles.CloseButton} flat onClick={onClose} data-id='modal-close'>
-                  Close <Close />
-                </Button>
+                <StyledCloseButton flat onClick={onClose} data-id="modal-close">
+                  <ScreenReaderOnly>Close</ScreenReaderOnly>
+
+                  <Close size={34} />
+                </StyledCloseButton>
               )}
+
               {children}
-            </Content>
-          </Grid.Column>
-        </Grid>
-      </div>
-    );
-  }
+            </div>
+          </ModalContent>
+        </StyledWrapper>
+      </StyledBase>
+    </Portal>
+  );
 }
 
+function ModalContent(props) {
+  const { open, children, maxWidth } = props;
+  const content = useRef(null);
+
+  return (
+    <StyledFocusLock disabled={!open} maxWidth={maxWidth}>
+      <Transition
+        mountOnEnter
+        unmountOnExit
+        in={open}
+        timeout={{
+          enter: 300,
+          exit: 150,
```

Comment:
Should this reference the fast token? I know the tokens aren't formatted in milliseconds – up to you, maybe add a comment here

Reply:
Actually wrote a helper for the app to use that does this. I'll just bring it in to Matchbox too.